### PR TITLE
NMR-5547: added new entrypoint for plugins MENU_FILE, and added new F…

### PR DESCRIPTION
**Pull request by JannalieT**
…ileExtensionType for txt files

Related to: https://github.com/nanalysis/nmrfx-private/pull/17

### CODE REVIEW

File: nmrfx-plugin-api/src/main/java/org/nmrfx/plugin/api/EntryPoint.java [hunk 1]
```diff
@@ -22,5 +22,5 @@
  * The obvious ones would be menus and scenes.
  */
 public enum EntryPoint {
-    STARTUP, MENU_PLUGINS, STATUS_BAR_TOOLS
+    STARTUP, MENU_PLUGINS, STATUS_BAR_TOOLS, MENU_FILE
```
Comment (vfiack) [inline nmrfx-plugin-api/src/main/java/org/nmrfx/plugin/api/EntryPoint.java:25]: small detail but I'd prefer if we kept the similar entrypoints closer together.

something like that:
```suggestion
    STARTUP, MENU_FILE, MENU_PLUGINS, STATUS_BAR_TOOLS
```

File: nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/FileMenuActions.java [hunk 2]
```diff
@@ -47,6 +49,8 @@ public void basic() {
         menu.getItems().addAll(openMenuItem, openDatasetMenuItem,
                 recentFIDMenuItem, recentDatasetMenuItem, datasetBrowserMenuItem,
                 graphicsMenu);
+
+        PluginLoader.getInstance().registerPluginsOnEntryPoint(EntryPoint.MENU_FILE, menu);
```
Comment (vfiack) [inline nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/FileMenuActions.java:53]: The other option was in makeMenuBar() in AnalystApp, but this works too.

**Review by vfiack — DISMISSED** [review]

File: nmrfx-plugin-api/src/main/java/org/nmrfx/plugin/api/EntryPoint.java (hunk 1 above)
Comment (JannalieT) [reply nmrfx-plugin-api/src/main/java/org/nmrfx/plugin/api/EntryPoint.java]: made this change

File: nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/FileMenuActions.java (hunk 2 above)
Comment (JannalieT) [reply nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/FileMenuActions.java]: moved to makeMenuBar

**Review by vfiack — APPROVED** [review]

**Pull request merged**